packages/python/plotly/plotly/basedatatypes.py

```diff
@@ -5597,12 +5601,24 @@ def to_plotly_json(self):
         """
         Return plotly JSON representation of object as a Python dict
 
+        Note: May include some JSON-invalid data types, use the `PlotlyJSONEncoder` util when encoding.
+
         Returns
         -------
         dict
         """
         return deepcopy(self._props if self._props is not None else {})
 
+    def to_json(self):
+        """
+        Convert to a JSON string representation.
+
+        Returns
+        -------
+        str
+        """
+        return json.dumps(self.to_plotly_json(), cls=PlotlyJSONEncoder)
```

Comment:
There's a better way to do this, using [`plotly.io.to_json`](https://plotly.com/python-api-reference/generated/plotly.io.html#plotly.io.to_json). The main difference is this will use `orjson` if it's installed, which can be much faster for large graphs.

Reply:
Ahhh cool, thanks! Updated 😄